src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -259,20 +264,23 @@ public static function get_theme_data( $deprecated = array(), $options = array()
 			$theme_json      = apply_filters( 'wp_theme_json_data_theme', new WP_Theme_JSON_Data( $theme_json_data, 'theme' ) );
 			$theme_json_data = $theme_json->get_data();
 			static::$theme   = new WP_Theme_JSON( $theme_json_data );
-		}
-
-		if ( wp_get_theme()->parent() ) {
-			// Get parent theme.json.
-			$parent_theme_json_data = static::read_json_file( static::get_file_path_from_theme( 'theme.json', true ) );
-			$parent_theme_json_data = static::translate( $parent_theme_json_data, wp_get_theme()->parent()->get( 'TextDomain' ) );
-			$parent_theme           = new WP_Theme_JSON( $parent_theme_json_data );
 
-			/*
-			 * Merge the child theme.json into the parent theme.json.
-			 * The child theme takes precedence over the parent.
-			 */
-			$parent_theme->merge( static::$theme );
-			static::$theme = $parent_theme;
+			if ( wp_get_theme()->parent() ) {
```

Comment:
This method, uses `wp_get_theme()` ALOT. This creates a new WP_Theme instance everytime. Settings a variable like this `$wp_theme = wp_get_theme();` would have some CPU cycles.

Reply:
Created a follow up issue - https://core.trac.wordpress.org/ticket/57057

Reply:
Thanks, will review it today!